router/h2/src/main/scala/io/buoyant/router/h2/ClassifierFilter.scala

```diff
@@ -11,17 +15,17 @@ object ClassifierFilter {
   val SuccessClassHeader = "l5d-success-class"
 
   def module: Stackable[ServiceFactory[Request, Response]] =
-    new Stack.Module1[param.ResponseClassifier, ServiceFactory[Request, Response]] {
+    new Stack.Module1[param.H2StreamClassifier, ServiceFactory[Request, Response]] {
       val role = ClassifierFilter.role
-      val description = "Sets the response classification into a header"
-      def make(classifierP: param.ResponseClassifier, next: ServiceFactory[Request, Response]) = {
-        val param.ResponseClassifier(classifier) = classifierP
+      val description = "Sets the stream classification into a header"
+      def make(classifierP: param.H2StreamClassifier, next: ServiceFactory[Request, Response]) = {
+        val param.H2StreamClassifier(classifier) = classifierP
         new ClassifierFilter(classifier).andThen(next)
       }
     }
 
-  val successClassClassifier: ResponseClassifier = {
-    case ReqRep(req, Return(rep: Response)) if rep.headers.contains(SuccessClassHeader) =>
+  val successClassClassifier: H2StreamClassifier = {
```

Comment:
I like your work on #1545 and I think that seems right – I'll happily make that change when we land #1545.